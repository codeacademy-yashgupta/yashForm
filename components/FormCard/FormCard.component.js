import React, { Component } from 'react';
import {
  Text, TouchableOpacity, View, StyleSheet,
} from 'react-native';
import style from './FormCard.style';

const styles = StyleSheet.create(style);
export default class FormCard extends Component {
  render() {
    const { id, title, createdAt } = this.props;
    return (
      <TouchableOpacity style={styles.formCard} key={id}>
        <Text style={styles.formTitle}>{title}</Text>
        <View
          style={styles.formHorizontalBar}
        />
        <Text style={styles.formCreatedAt}>
          Created At
          {' '}
          {createdAt}
        </Text>
      </TouchableOpacity>
    );
  }
}
