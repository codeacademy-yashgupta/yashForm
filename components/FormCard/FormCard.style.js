import { APP_FONT_SIZE, APP_BODY_TEXT_COLOR } from '../../constants';

export default {
  formCard: {
    height: 200, width: 300, backgroundColor: 'white', margin: 20, padding: 20,
  },
  formTitle: {
    fontSize: APP_FONT_SIZE, fontWeight: 'bold', color: APP_BODY_TEXT_COLOR, height: 100,
  },
  formHorizontalBar: {
    borderBottomColor: 'black', borderBottomWidth: 1, marginTop: 20, marginBottom: 20,
  },
  formCreatedAt: {
    color: APP_BODY_TEXT_COLOR, right: 0, position: 'absolute', bottom: 0, marginRight: 15, marginBottom: 10,
  },
};
