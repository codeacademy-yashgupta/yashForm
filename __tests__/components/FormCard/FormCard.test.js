import React from 'react';
import renderer from 'react-test-renderer';
import FormCard from '../../../components/FormCard/FormCard.component';
import 'react-native';

describe('FormCard', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<FormCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
