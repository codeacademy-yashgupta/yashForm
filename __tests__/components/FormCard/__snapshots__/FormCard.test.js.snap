// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard should render correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "height": 200,
      "margin": 20,
      "opacity": 1,
      "padding": 20,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "height": 100,
      }
    }
  />
  <View
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  />
  <Text
    style={
      Object {
        "bottom": 0,
        "color": "black",
        "marginBottom": 10,
        "marginRight": 15,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    Created At
     
  </Text>
</View>
`;
